apps/barista/src/pages/search-page/search-page.ts

```diff
@@ -0,0 +1,109 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaPage } from '../page-outlet';
+import {
+  Component,
+  Input,
+  ViewChild,
+  ElementRef,
+  OnInit,
+  OnDestroy,
+  AfterViewInit,
+} from '@angular/core';
+import {
+  BaSinglePageContent,
+  BaSearchResult,
+} from '@dynatrace/barista-components/barista-definitions';
+import { Observable, fromEvent, merge, Subject } from 'rxjs';
+import {
+  debounceTime,
+  map,
+  filter,
+  distinctUntilChanged,
+  takeUntil,
+  concatMap,
+  tap,
+  take,
+} from 'rxjs/operators';
+import { BaSearchService } from '../../shared/search.service';
+import { BaLocationService } from '../../shared/location.service';
+
+@Component({
+  selector: 'ba-search-page',
+  templateUrl: 'search-page.html',
+  styleUrls: ['search-page.scss'],
+})
+export class BaSearchPage implements BaPage, AfterViewInit, OnDestroy {
+  /** Destroy subject. */
+  private _destroy$ = new Subject<void>();
+
+  /**
+   * Contents input that is necessary for the BaPage interface,
+   * but is not in use
+   */
+  @Input() contents: BaSinglePageContent;
+
+  /** @internal Viewchild of the query input field to get the changes events. */
+  @ViewChild('queryInput', { static: true }) _queryInput: ElementRef;
+
+  /** @internal Search results for the last query. */
+  _searchResults$: Observable<BaSearchResult[]>;
+
+  /** Value of the query mapped to the input. */
+  queryValue: string;
+
+  constructor(
+    private readonly _searchService: BaSearchService,
+    private readonly _locationService: BaLocationService,
+  ) {}
+
+  ngAfterViewInit(): void {
+    // Get the current query from the route and set it into the search
+    // field if necessary.
+    const currentRouteQuery = this._locationService.currentQuery$.pipe(
+      map(params => params.get('q')),
+      filter(query => query !== null),
+      take(1),
+      tap(query => (this.queryValue = query!)),
+    );
+
+    // Get updates from the search input field
+    // and update the location and searchresults if necessary
+    const querySearchQuery = fromEvent(
+      this._queryInput.nativeElement,
+      'input',
+    ).pipe(
+      debounceTime(100),
+      map(() => this.queryValue),
+      filter((query: string) => query.length > 0),
+      distinctUntilChanged(),
+      tap(query => this._locationService.go(`/search/?q=${query}`)),
+    );
+
+    // Subscribe to the current query and
+    // query the search api.
+    this._searchResults$ = merge(currentRouteQuery, querySearchQuery).pipe(
+      takeUntil(this._destroy$),
+      concatMap((query: string) => this._searchService.search(query)),
```

Comment:
please use `switchMap` concatMap does not cancel the previous stream it will wait til this completes until it subscribes to the new. In this case, I would cancel the previous and immediately subscribe to the new stream.